Symfony/CS/Tokenizer/Tokens.php

```diff
@@ -1083,9 +1085,11 @@ public function toJson()
      */
     public function __clone()
     {
-        foreach ($this as $key => $val) {
-            $this[$key] = clone $val;
+        foreach ($this as $key => &$val) {
+            $val = clone $val;
         }
+        // unset reference to keep scope clear
+        unset($val);
```

Comment:
don't like that, could you elaborate why referencing and extra unset is better ?